src/main/java/ui/components/pickers/LabelPickerDialog.java

```diff
@@ -3,157 +3,282 @@
+                                       Optional<String> suggestion) {
+        initialLabels.stream()
+            .forEach(label -> assignedLabels.getChildren()
+            .add(processInitialLabel(label, removedLabels, suggestion)));
+    }
+
+    private Node processInitialLabel(String initialLabel, List<String> removedLabels, 
+                                            Optional<String> suggestion) {
+        TurboLabel repoInitialLabel = getRepoTurboLabel(initialLabel);
+        if (!removedLabels.contains(initialLabel)) {
+            if (suggestion.isPresent() && initialLabel.equals(suggestion.get())) {
+                return getStyledPickerLabel(repoInitialLabel, true, false, true, false, true);
             }
-        });
+            return getStyledPickerLabel(repoInitialLabel, false, false, false, false, true);
+        }
+
+        if (suggestion.isPresent() && initialLabel.equals(suggestion.get())) {
+            return getStyledPickerLabel(repoInitialLabel, true, false, false, false, true);
+        }
+
+        return getStyledPickerLabel(repoInitialLabel, false, false, true, false, true);
     }
 
-    @SuppressWarnings("unused")
-    private void ______POPULATION______() {}
+    private Node getStyledPickerLabel(TurboLabel label, 
+        boolean isFaded, boolean isHighlighted, boolean isRemoved, boolean isSelected, boolean isTop) {
+        PickerLabel styledLabel = new PickerLabel(label, isTop);
+        styledLabel.setIsFaded(isFaded);
+        styledLabel.setIsHighlighted(isHighlighted);
+        styledLabel.setIsRemoved(isRemoved);
+        styledLabel.setIsSelected(isSelected);
 
-    protected void populatePanes(List<PickerLabel> existingLabels, List<PickerLabel> newTopLabels,
-            List<PickerLabel> bottomLabels, Map<String, Boolean> groups) {
-        populateTopPane(existingLabels, newTopLabels);
-        populateBottomBox(bottomLabels, groups);
+        Node node = styledLabel.getNode();
+        node.setOnMouseClicked(e -> handleLabelClick(styledLabel));
+        return node;
     }
 
-    private void populateTopPane(List<PickerLabel> existingLabels, List<PickerLabel> newTopLabels) {
-        topPane.getChildren().clear();
-        if (existingLabels.isEmpty() && newTopLabels.isEmpty()) {
-            Label label = new Label("No currently selected labels. ");
-            label.setPadding(new Insets(2, 5, 2, 5));
-            topPane.getChildren().add(label);
-        } else {
-            existingLabels.forEach(label -> topPane.getChildren().add(label.getNode()));
-            if (!newTopLabels.isEmpty()) {
-                topPane.getChildren().add(new Label("|"));
-                newTopLabels.forEach(label -> topPane.getChildren().add(label.getNode()));
-            }
+    /**
+     * To return the repo's TurboLabel that contains labelName
+     * <p>
+     * labelName MUST match a repo label's name
+     *
+     * @param labelName
+     * @return
+     */
```

Comment:
[1] How about this?
```java
/**
 * Returns the TurboLabel that matches the labelName. 
 * Assumption: the labelName matches exactly one TurboLabel
*/
```
[2] This can be a static method in `TurboLabel` class. `getTurboLabel(List<TurboLabel> labels, String labelName)`